pkgs/tools/misc/nix-direnv/default.nix

```diff
@@ -14,9 +14,8 @@ stdenv.mkDerivation rec {
   # Substitute instead of wrapping because the resulting file is
   # getting sourced, not executed:
   postPatch = ''
-    substituteInPlace direnvrc \
-      --replace "\''${NIX_BIN_PREFIX:-}" "\''${NIX_BIN_PREFIX:-${nix}/bin/}" \
-      --replace "grep" "${gnugrep}/bin/grep"
+    sed -i "1a NIX_BIN_PREFIX=${nix}/bin/" direnvrc
```

Comment:
On the second thought it might be actually a bad idea to use nix stable here because it will
prevent nixFlakes user to use `use flake` as it hard codes the nix version. I don't see any downside into the other direction.

Reply:
What does that mean concrete? Using `nixFlakes` instead? 

IMHO the old approach with the replacement would avoid this as it searches for the correct nix implementation and only if it is not found, it will use the `nix` package as fallback, right?

Reply:
What does that mean concrete? Using `nixFlakes` instead?

IMHO the old approach with the replacement would avoid this as it searches for the correct `nix` implementation and only if it is not found, it will use the `nix` package as fallback, right?

Reply:
Yes. `nixFlakes` should be used.

Reply:
I currently don't see a use case for fallbacks. If you want to override nix, than you can do so by overriding it in the inputs of this derivation.

Reply:
> Yes. `nixFlakes` should be used.
Honestly, I don't understand the benefits and drawbacks of either solution but yes, I can override it with `nix-build -I nixpkgs=$(pwd) -E "with (import <nixpkgs>{}); nix-direnv.override { nixFlakes = pkgs.nix; }"` and it uses my normal `nix` version.


> I currently don't see a use case for fallbacks. If you want to override nix, than you can do so by overriding it in the inputs of this derivation.
Ok